app/api/deps/webhooks/zalo_webhook_deps.py

```
@@ -0,0 +1,86 @@
+from fastapi import Header, Request, HTTPException
+import hmac
+import hashlib
+
+from app.core.config import settings
+
+
+def build_zalo_webhook_signature_content(data: dict) -> str:
+    """
+    Build the string Zalo hashes for x-zevent-signature (before + secret key).
+
+    Spec: sort top-level field names A–Z, append each field's value in that
```

Ruff (0.15.11)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)